ci-operator/step-registry/azure/provision/sharednetwork/azure-provision-sharednetwork-commands.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```

Comment:
I think we do not need this new ref, but we can improve the logic based on `ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh`.
Change the logic
```
cat > "${PATCH}" << EOF
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
EOF
```
to 
```
if [ -f "${SHARED_DIR}/customer_vnet_subnets.yaml" ]; then
  <blablabla>
else
  cat > "${PATCH}" << EOF
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
EOF
fi
```